web/server/watch/integration.go

```diff
@@ -43,91 +43,90 @@ func NewWatcher(rootFolder string, folderDepth int, nap time.Duration,
 	}
 }
 
-func (this *Watcher) Listen() {
+func (w *Watcher) Listen() {
 	for {
-		if this.stopped {
-			return
+		if !w.paused {
+			w.scan()
 		}
-
 		select {
-
-		case command := <-this.input:
-			this.respond(command)
-
-		default:
-			if !this.paused {
-				this.scan()
+		case command := <-w.input:
+			if stopped := w.respond(command); stopped {
+				return
 			}
-			time.Sleep(this.nap)
```

Comment:
the part above was very likely to be nondeterministic in tests. Consider when two commands show up on the this.input channel in short succession. Sometimes the code here would pull both of those commands without a call to the "default" in the middle. Sometimes you'd get a command pulled, then a scan (and sleep), then the other command. Those made the tests run very flakily. The new way: (1) guarantees a scan before each new command is pulled and (2) stops sleeping when there's a command to be pulled